### ISSUE TITLE
twisted.conch.test.test_ssh.UtilityTestCase.testCounter relies on deprecated wrapping behavior of struct module

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2057]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2057|
|Type|defect|
|Created|2006-09-02 20:26:41Z|

It passes in values above 65535 for an 'H' data item and expects the value to be wrapped back to 0.

<details><summary>Searchable metadata</summary>

```
trac-id__2057 2057
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1157228801000000 1157228801000000
changetime__1157565327000000 1157565327000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

```
Index: twisted/conch/test/test_ssh.py
===================================================================
--- twisted/conch/test/test_ssh.py      (revision 18030)
+++ twisted/conch/test/test_ssh.py      (working copy)
@@ -249,10 +249,10 @@
         def testCounter(self):
             c = transport._Counter('\x00\x00', 2)
             for i in xrange(256 * 256):
-                self.assertEquals(c(), struct.pack('!H', i + 1))
+                self.assertEquals(c(), struct.pack('!H', (i + 1) % (2 ** 16)))
             # It should wrap around, too.
             for i in xrange(256 * 256):
-                self.assertEquals(c(), struct.pack('!H', i + 1))
+                self.assertEquals(c(), struct.pack('!H', (i + 1) % (2 ** 16)))


     class ConchTestPublicKeyChecker(checkers.SSHPublicKeyDatabase):
```

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_2"></a>|@jerub set owner to @exarkun|
|-|-|

I had to deal with wrapping signed integer values in pypy just the other day. Please let me know if you know anything more sane than: ((x + 2**31) % 2**32)-2**31 to do that...

Anyway, looks good, please do that merge thing.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [18059]) Apply patch from [[#2057](https://github.com/twisted/twisted/issues/2057)](https://github.com/twisted/twisted/issues/2057) - explicitly wrap 16 bit overflow values

Author: exarkun
Reviewer: jerub
Fixes #2057

Python 2.5's struct module emits a deprecation warning when a
value greater than or equal to 2 ** 16 is passed for an "H"
format element.  The desired behavior in this conch test is for
it to be wrapped around, so we'll wrap it explicitly.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
